### PR TITLE
Use LoadingCache for isolation of CmServerClusterInstance.cluster

### PR DESCRIPTION
**Pull request by abayer**

### CODE REVIEW

File: src/main/java/com/cloudera/whirr/cm/CmServerClusterInstance.java [hunk 1]
```diff
@@ -120,27 +133,29 @@ public static synchronized CmServerFactory getFactory(CmServerFactory factory) {
...
       SortedSet<String> mounts) throws CmServerException, IOException {
-    return getCluster(configuration, instances, mounts, Collections.<String> emptySet());
+    return getCluster(spec, configuration, instances, mounts, Collections.<String> emptySet());
   }
 
-  public static synchronized CmServerCluster getCluster(Configuration configuration, Set<Instance> instances,
+  public static synchronized CmServerCluster getCluster(ClusterSpec spec, Configuration configuration, Set<Instance> instances,
       SortedSet<String> mounts, Set<String> roles) throws CmServerException, IOException {
-    cluster = new CmServerCluster();
+    CmServerCluster cluster = new CmServerCluster();
```
Comment (ggear) [inline src/main/java/com/cloudera/whirr/cm/CmServerClusterInstance.java:158]: Like for like change here would also ensure the cluster is cached, ie

``` java
    CmServerCluster cluster = new CmServerCluster();
    clusterCache.put(new Key(spec), cluster);
```

Comment (abayer) [inline src/main/java/com/cloudera/whirr/cm/CmServerClusterInstance.java:158]: I'll give that a shot.

File: src/main/java/com/cloudera/whirr/cm/CmServerClusterInstance.java [hunk 2]
```diff
@@ -46,23 +46,36 @@
...
 
   private static CmServerFactory factory;
-  private static CmServerCluster cluster;
   private static boolean isStandaloneCommand = true;
   private static Map<ClusterActionEvent, Set<Integer>> ports = new HashMap<ClusterActionEvent, Set<Integer>>();
 
   private CmServerClusterInstance() {
   }
 
+  static final LoadingCache<Key, CmServerCluster> clusterCache = CacheBuilder.newBuilder().build(
```
Comment (ggear) [inline src/main/java/com/cloudera/whirr/cm/CmServerClusterInstance.java]: This should be private, ie:

``` java
private static final LoadingCache<Key, CmServerCluster> clusterCache = CacheBuilder.newBuilder().build(
```

**Comment by ggear** [comment]
Can we add a unit test?

**Comment by abayer** [comment]
Unit test added, along with the changes you requested.

**Pull request merged**